### DISCUSSION TITLE
NameError: name 'RotaryEncoder' is not defined

### DISCUSSION THREAD

**Discussion by allcoolthingsatoneplace** [gpiozero/gpiozero#993] [Q&A]
Operating system: debian 10.9
Python version: Python 2.7.16
Pi model: Raspberry Pi 3 Model B+
GPIO Zero version:  python-gpiozero (1.6.2-1)
Pin factory used: RPiGPIO
See http://rpf.io/gpzissue for information on how to find out these details

Please give us as much information about your issue as possible. Write your
code inside code blocks like so:

```python
from gpiozero import RotaryEncoder

# GPIO Pins #
Enc_A = 17  
Enc_B = 27
#-----------#
Enc_C = 22  
Enc_D = 23
#-----------#
Enc_E = 24  
Enc_F = 25
#-----------#

rEncoder1 = RotaryEncoder(Enc_A, Enc_B, wrap=True, max_steps=180)
rEncoder2 = RotaryEncoder(Enc_C, Enc_D, wrap=True, max_steps=180)
rEncoder3 = RotaryEncoder(Enc_E, Enc_F, wrap=True, max_steps=180)

rEncoder1.steps = -180
rEncoder2.steps = -180
rEncoder3.steps = -180

def show_rEncoder1_value():
    print(rEncoder1.values)

def show_rEncoder2_value():
    print(rEncoder2.values)

def show_rEncoder3_value():
    print(rEncoder3.values)

rEncoder1.when_rotated = show_rEncoder1_value
rEncoder2.when_rotated = show_rEncoder2_value
rEncoder3.when_rotated = show_rEncoder3_value
```

**Comment by bennuttall** [comment]
That error doesn't make sense. If you were using a version before RotaryEncoder was added, it would be an ImportError. A NameError means you've referred to it before it was defined. But that's not the case in the code you've provided.

It's possible you ran some code before importing RotaryEncoder, or after ignoring an ImportError if the Python shell (which would explain the NameError).

Try saving the *exact* code above as a file and running it with `python file.py`.

Is there a reason you're using Python 2? It's been EOL for nearly two years now. You should switch to Python 3.

There are two Debian packages for gpiozero: python-gpiozero (for Python 2) and python3-gpiozero (for Python 3). Make sure you have both installed. If you're using Debian instead of Raspberry Pi OS, then Debian only has an old version of gpiozero (before Rotary Encoder was added).

Also is there a reason you're using Debian, not Raspberry Pi OS? You might find it easier to use the supported OS in which things tend to work out of the box.

**Reply by allcoolthingsatoneplace** [reply]
im using ( Raspberry Pi OS -> Raspbian or Debian 10.9 ) i created it from rpi imager [https://downloads.raspberrypi.org/imager/imager_latest.exe](url)
i have both versions of python and python3 installed with both gpiozero packages, installed with pip, apt-get and apt ways...
this method of importing
`from gpiozero import RotaryEncoder`
and also this one
`import gpiozero`
and no luck.
and yeah im using exactly the same code as mentioned above.
thanks.

**Reply by bennuttall** [reply]
I need to see a full stack trace
